StripePaymentSheet/StripePaymentSheet/PaymentSheet/Intent.swift

```diff
@@ -199,18 +201,28 @@ extension STPConfirmPaymentMethodOptions {
         paymentMethodType: STPPaymentMethodType,
         customer: PaymentSheet.CustomerConfiguration?
     ) {
-        // This property cannot be set if there is no customer.
+        // Something went wrong if we're trying to save and there's no Customer!
         assert(!(shouldSave && customer == nil))
 
-        // Only support card and US bank setup_future_usage in payment_method_options
-        guard customer != nil && paymentMethodType == .card || paymentMethodType == .USBankAccount
-        else {
+        // This property should only be set if
+        // 1. We're displaying a "Save this pm for future payments" checkbox
+        // 2. The PM type is card or US bank
+        guard customer != nil && paymentMethodType == .card || paymentMethodType == .USBankAccount else {
+            return
+        }
+        // Passing "none" *overrides* the top-level setup_future_usage and is not what we want, b/c this code is called even when we don't display the "save" checkbox (e.g. when the PI top-level setup_future_usage is already set).
+        // Instead, we pass an empty string to 'unset' this value. This makes the PaymentIntent *inherit* the top-level setup_future_usage.
```

Comment:
Similarly, might make sense to put this in the doc string for the `paymentMethodType` param

Reply:
This one is an implementation detail attempting to describe why use "" instead of "none", so I'll leave it here but maybe try to reword a bit.  It's very complicated :(